hopsworks-common/src/main/java/io/hops/hopsworks/common/util/Settings.java

```diff
@@ -406,6 +411,40 @@ private void checkCache() {
     }
   }
 
+  public synchronized void refreshCache() {
+    cached = false;
+    populateCache();
+  }
+  
+  public synchronized void updateVariable(String variableName, String variableValue)
+    throws IOException {
+    try {
+      updateVariableInternal(variableName, variableValue);
+      refreshCache();
+    } catch (NoResultException ex) {
+      logger.log(Level.WARNING, "Error while updating variable: " + variableName, ex);
+      throw new IOException(ex);
+    }
+  }
+  
+  public synchronized void updateVariables(Map<String, String> variablesToUpdate)
+    throws IOException {
+    Map<String, String> rollBackVariables = new HashMap<>(variablesToUpdate.size());
```

Comment:
@kouzant @ErmiasG can we do a rollback by using J2EE transactions? Something like this https://github.com/hopshadoop/hopsworks/blob/65aa3c88da2fdfab18aca755fbe0a8687b9f81b3/hopsworks-admin/src/main/java/io/hops/hopsworks/admin/user/account/ResetPassword.java#L177

Reply:
The problem is that in https://github.com/hopshadoop/hopsworks/blob/master/hopsworks-common/src/main/java/io/hops/hopsworks/common/util/Settings.java we catch the NoResultException and just return `null`, so I'm not sure if the transaction will be rolled back. That's the assumption for the rest of the operations and I didn't want to change this.

Reply:
It's ok for now, we must do a review of the code in different places where we rollback in the application level.

Reply:
Used Transaction rollback instead of manual.